### PR TITLE
Implemented IsConstant Module

### PR DESCRIPTION
**Pull request by Smit1603**
- Fixes compiler-research/InterOp#13
- I have implemented and successfully tested IsConstant Module
- However there has been an unexpected hardware issue on my device so it has to be sent for servicing purpose, so it will be difficult for me to take up and resolve your issues/suggestions immediately
- Kindly review and pls let me know suggestions/improvements
- I would try to resolve and look over your suggestion as soon as possible 
- I am currently finding a way to continue being active and contribute

### CODE REVIEW

File: lib/Interpreter/InterOp.cpp [hunk 1]
```diff
@@ -2267,5 +2267,22 @@ namespace InterOp {
...
+
+    auto *D = (clang::Decl *)method;
+    auto *FD = llvm::dyn_cast_or_null<clang::FunctionDecl>(D);
+
+    if (!FD->isCXXClassMember())
+      return false;
+      
+    const auto func = dyn_cast<CXXMethodDecl>(FD);
+    if (func && func->getMethodQualifiers().hasConst())
+      return true;
```
Comment (vgvassilev) [inline lib/Interpreter/InterOp.cpp:2282]:
```suggestion
    if (!method)
      return false;
    auto *D = (clang::Decl *)method;

    if (auto *func = dyn_cast<CXXMethodDecl>(D))
       return func->getMethodQualifiers().hasConst());
```

**Comment by vgvassilev** [comment]
Please consider linking this PR with the opened issue in the commit message: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by Smit1603** [comment]
- @vgvassilev extremely sry for delay
- I have made recommended changes
- Pls review and let me know whether any other changes/Improvements are required or not

File: lib/Interpreter/InterOp.cpp [hunk 2]
```diff
@@ -2267,5 +2267,27 @@ namespace InterOp {
...
+
+    // auto *D = (clang::Decl *)method;
+    // auto *FD = llvm::dyn_cast_or_null<clang::FunctionDecl>(D);
+
+    // if (!FD->isCXXClassMember())
+    //   return false;
+      
+    // const auto func = dyn_cast<CXXMethodDecl>(FD);
+    // if (func && func->getMethodQualifiers().hasConst())
+    //   return true;
```
Comment (vgvassilev) [inline lib/Interpreter/InterOp.cpp:2282]: Please delete the unused code. Also make all of this in one commit with a good commit message.

File: lib/Interpreter/InterOp.cpp [hunk 3]
```diff
@@ -2267,5 +2267,27 @@ namespace InterOp {
...
+    if (!method)
+      return false;
+    auto *D = (clang::Decl *)method;
```
Comment (sudo-panda) [inline lib/Interpreter/InterOp.cpp:2285]:
```suggestion
      return false;

    auto *D = (clang::Decl *)method;
```

**Review by sudo-panda — CHANGES_REQUESTED** [review]

**Comment by Smit1603** [comment]
- @vgvassilev 
- I have made all recommended changes
- I am facing similar issue as my last Pull Request(merge conflict) as you might know
- I am sorry I could not resolve this issue immediately, but is it possible for you to explain me how to resolve this issue on git so it might be a good learning experience for me as well as it will help me in future
- Sry for asking this silly question

**Comment by vgvassilev** [comment]
I have fixed it.

**Pull request merged**